sklearn/kernel_ridge.py

```diff
@@ -48,11 +48,12 @@ class KernelRidge(MultiOutputMixin, RegressorMixin, BaseEstimator):
         assumed to be specific to the targets. Hence they must correspond in
         number. See :ref:`ridge_regression` for formula.
 
-    kernel : str or callable, default="linear"
+    kernel : {'additive_chi2', 'chi2', 'cosine', 'linear', 'poly', \
```

Comment:
I think that we did not want this change in the past because there are too many possibilities.
That's why we specified the constant `pairwise.PAIRWISE_KERNEL_FUNCTIONS`

Reply:
Good point, I will revert this and add a commit that specifies the allowance of `precomputed`.